Mergin/version_viewer_dialog.py

```diff
@@ -0,0 +1,639 @@
+        self.model_detail.setHorizontalHeaderLabels(["Details"])
+
+        self.details_treeview.setEditTriggers(QAbstractItemView.NoEditTriggers)
+        self.details_treeview.setModel(self.model_detail)
+
+        self.versionModel.current_version = self.mp.version()
+
+    def exec(self):
+
+        try:
+            ws_id = self.mp.workspace_id()
+        except ClientError as e:
+            QMessageBox.warning(None, "Client Error", str(e))
+            return
+
+        # check if user has permissions
+        try:
+            usage = self.mc.workspace_usage(ws_id)
+            if not usage["view_history"]["allowed"]:
+                QMessageBox.warning(
+                    None, "Upgrade required", "To view the project history, please upgrade your subscription plan."
+                )
+                return
+        except ClientError:
+            # Some versions e.g CE, EE edition doesn't have
+            pass
+        super().exec()
+
+    def closeEvent(self, event):
+        self.save_splitters_state()
+        QDialog.closeEvent(self, event)
+
+    def save_splitters_state(self):
+        settings = QSettings()
+        settings.setValue("Mergin/VersionViewerSplitterSize", self.splitter_map_table.saveState())
+        settings.setValue("Mergin/VersionViewerSplitterVericalSize", self.splitter_vertical.saveState())
+
+    def set_splitters_state(self):
+        settings = QSettings()
+        state_vertical = settings.value("Mergin/VersionViewerSplitterVericalSize")
+        if state_vertical:
+            self.splitter_vertical.restoreState(state_vertical)
+        else:
+            self.splitter_vertical.setSizes([120, 200, 40])
+
+        do_calc_height = True
+        state = settings.value("Mergin/VersionViewerSplitterSize")
+        if state:
+            self.splitter_map_table.restoreState(state)
+
+            if self.splitter_map_table.sizes()[0] != 0:
+                do_calc_height = False
+
+        if do_calc_height:
+            height = max([self.map_canvas.minimumSizeHint().height(), self.attribute_table.minimumSizeHint().height()])
+            self.splitter_map_table.setSizes([height, height])
+
+    def fetch_from_server(self):
+
+        if self.fetcher and self.fetcher.isRunning():
+            # Only fetching when previous is finshed
+            return
+        else:
+            self.fetcher.start()
+
+    def on_scrollbar_changed(self, value):
+
+        if self.ui.history_treeview.verticalScrollBar().maximum() <= value:
+            self.fetch_from_server()
+
+    def current_version_changed(self, current_index, previous_index):
+        # Update the ui when the selected version change
+        item = self.versionModel.item_from_index(current_index)
+        version_name = item["name"]
+        version = int_version(item["name"])
+
+        self.setWindowTitle(f"Changes Viewer | {version_name}")
+
+        self.version_details = self.mc.project_version_info(self.mp.project_id(), version_name)
```

Comment:
Yes the key are guaranteed
And if the user doesn't have access to internet when openning the project the icon are grayed out like that: 
![Screenshot From 2024-12-12 12-27-16](https://github.com/user-attachments/assets/cdea3c9d-adc3-4902-897d-7b749cb9a52a)

**Howewer if the user disconnect from internet in the middle of the use of the plugin it would have failed with an error, I added an extra try/catch to prevent it**